pkg/internal/coordinatortest/coordinatortest.go

```diff
@@ -0,0 +1,193 @@
+// Package coordinatortest contains scenarios meant to be used for pkg/coordinator
```

Comment:
From [Go docs](https://go.dev/doc/go1.4#internalpackages):

> When the go command sees an import of a package with internal in its path, it verifies that the package doing the import is within the tree rooted at the parent of the internal directory. For example, a package .../a/b/c/internal/d/e/f can be imported only by code in the directory tree rooted at .../a/b/c. It cannot be imported by code in .../a/b/g or in any other repository.

I think we should place this code in `pkg/coordinator/internal`, closer to the code it is testing.

`pkg/internal` should contain only internal code used by more than one package.